sql/dj_setup_database.in

```diff
@@ -354,6 +370,64 @@ upgrade)
 	verbose "DOMjudge database upgrade completed."
 	;;
 
+snapshot-dump)
+	read_dbpasswords
+	if [ -z "$DUMPNAME" ]; then
+		usage
+		exit 1
+	fi
+
+	if [ -f "${DATABASEDUMPDIR}/${DUMPNAME}.sql.gz" ]; then
+		while true; do
+			read -p "Overwrite existing database dump (y/N)? " yn
+			case $yn in
+				[Yy]* ) create_database_dump $DUMPNAME ;;
+				''|[Nn]* ) break;;
+			esac
+		done
+	else
+		create_database_dump $DUMPNAME
+	fi
+
+	read -p "Provide prefix or say 'no' to abort. Format: [prefix]-$1 " prefix
```

Comment:
This should just be done if there was already a dump and we decided not to overwrite, right?

Reply:
We might even want to consider not asking the question and backing the old dump up with a timestamp (and writing a warning that we did this)

Reply:
I agree. I think the current flow is a bit confusing, both to read in code and as a user. I think that if you specify something that already exists, then you either can accept to overwrite it or don't and then the script will abort. We shouldn't offer a prefix to fix that inline. Just run the script again with a different argument.
@meisterT 's suggestion would be an alternative, but then it's also implicit that the original dump gets renamed.

Reply:
I went with the easier option of @eldering.